2018/002-update-channels.md

```diff
@@ -0,0 +1,145 @@
+ Title
+
+| Field  | Value      |
+|:-------|:-----------|
+| Status | Draft      |
+| Date   | 2018-07-31 |
+
+## Introduction
+
+### Problem description
+
+Updates can be disruptive for a customer's application stack (ex. Kubernetes
+upgrade from 1.10.x to 1.11.x, default pod security policy introduction).
+
+Right now there is no visibility of what is inside an update and furthermore
+there is no possible way around intrusive updates by having the ability to only
+apply critical bug fixes and security updates. An admin needs to apply all updates
+that come through a single channel.
+
+In order to separate intrusive updates in SUSE CaaS Platform, we need different
+update channels. This allows the separation of possibly disruptive updates
+into their own channels, and lets the admin continue applying bugfixes and security
+updates without the need to apply intrusive, new features. The admin can also see
+what is inside an update to be able to decide when to apply it.
+
+The channels will also act as a barrier to avoid that a user runs k8s 1.8 and
+upgrades right away to 1.11 in one step.
+
+### Proposed change
+
+The admin will have the ability of choosing when to switch to the next channel.
+
+In case of an airgapped system, a link to the release notes of the update channel
+is displayed. A system with access to the web can fetch the release notes and
+display them in the UI directly. In both scenarios, an admin can evaluate the
+upcoming changes beforehand.
+
+The UI of velum needs to be adapted to enable a switch to a new channel.
+
+## Detailed RFC
+
+We can separate the updates into different channels. There are 2 types of channels:
+
+* update channel: normal maintenance updates, which are safe and non-intrusive
+* new version channel: version updates which can contain intrusive updates
```

Comment:
I'm a bit confused, maybe it's the wording here or it's just me...

I understood CaaS Platform installations will still have one single repository active. This repository (aka channel) is associated with a certain version of the product, for example: `3.1`. The channel is going to provide regular maintenance updates (eg: security fixes) and bug fixes for the specific product codebase associated with the channel.

Invasive updates are going to be shipped to a new different channel. This will result in a minor version of the product to be announced.

Once a new channel is published customers will have a choice between moving to the new one, or stay on the current one. For X months the old channel will get security fixes and bug fixes for the product that do not require intrusive changes. Once this time is over the customer can stay registered against the channel but he won't get any update at all (even security fixes); plus he won't be supported.


To make a concrete example:
1. The current version of CaaSP is 3.0
    1. `bash` security fix is sent to the 3.0 channel
    1. `kubernetes-salt` bug fixes are sent to the 3.0 channel
1. A new version of Kubernetes is released:
    1. The 3.1 channel is created
    1. A ssl security fix is pushed both to the 3.0 and to the 3.1 channel
    1. `kubernetes-salt` bug fixes are sent to the 3.1 channel and eventually to the 3.0 one
1. *X* months passed since the creation of the 3.1 channel
    1. A kernel security fix is pushed **only** to the 3.1 channel

Did I get it right?

Reply:
SUSE CaaS Platform has always two active channels: the update and the pool channel. Both have the same version. This will stay. 
Else what you wrote is correct, the "update channel" is the current "3.0" update channel in your example, the "new version channel" is the new "3.1 update channel" from your example.

Reply:
Based on what has been discussed with @thkukuk, I think we have to change the following lines:

> There are 2 types of channels:
> * update channel: normal maintenance updates, which are safe and non-intrusive
> * new version channel: version updates which can contain intrusive updates

It is not true that we have two updates channels per release. The update channel will stay the same: security fixes and intrusive changes are going to be pushed to the same channel.